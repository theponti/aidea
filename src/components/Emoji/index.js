import React from 'react'
import t from 'prop-types'
import cx from 'classnames'
import styles from './Emoji.module.scss'

interface EmojiProps {
  className?: string;
  label: string;
  emoji: string;
}

Emoji.propTypes = {
  className: t.string,
  label: t.string,
  emoji: t.string
}

function Emoji ({ className = '', label, emoji, ...props }: EmojiProps) {
  return (
    <span
      role="img"
      className={cx(styles.container, className)}
      aria-label={label}
      {...props}
    >
      {emoji}
    </span>
  )
}

export default Emoji
