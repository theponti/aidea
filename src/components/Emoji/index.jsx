import React from "react";
import t from "prop-types";
import cx from "classnames";
import styles from "./Emoji.module.scss";

function Emoji({ className = "", label, emoji, ...props }) {
  return (
    <span
      role="img"
      className={cx(styles.container, className)}
      aria-label={label}
      {...props}
    >
      {emoji}
    </span>
  );
}

Emoji.propTypes = {
  className: t.string,
  label: t.string,
  emoji: t.string,
};

export default Emoji;
