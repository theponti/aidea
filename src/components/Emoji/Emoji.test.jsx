import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Emoji from ".";

describe("<Emoji/>", () => {
  it("should render component", () => {
    const { container } = render(<Emoji label="sheep" emoji="🐑" />);
    expect(container).toMatchSnapshot();
  });
});
