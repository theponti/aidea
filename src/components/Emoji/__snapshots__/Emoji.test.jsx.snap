// Vitest Snapshot v1

exports[`<Emoji/> > should render component 1`] = `
<div>
  <span
    aria-label="sheep"
    class="_container_11i26_1"
    role="img"
  >
    🐑
  </span>
</div>
`;
